lib/files.js

```diff
@@ -0,0 +1,54 @@
+const { checkIfPathMatches } = require('./gitpatternParser')
+
+/**
+ * process the PR based on files settings
+ *
+ * @return
+ *  JSON object with the properties mergeable and the description if
+ *  `mergeable` is false
+ */
+module.exports = async (pr, context, settings) => {
+  let fileSettings = settings.files
+  if (!fileSettings) {
+    return {
+      mergeable: true,
+      description: 'Okay To Merge'
+    }
+  }
+
+  let patternToMatch = settings.files.pattern
+  // fetch the file list
+  let result = await context.github.pullRequests.getFiles(context.repo({number: pr.number}))
+  let addedFiles = result.data.filter(file => file.status === 'added' || file.status === 'modified')
+  // parse the files to match the pattern
+  const filesToCheck = addedFiles.filter(file => checkIfPathMatches(file.filename, patternToMatch))
+
+  // fetch the file content
+  result = await fetchFileContents(filesToCheck, context)
+  const failedFiles = result.filter(file => {
```

Comment:
I noticed you've been using `const` and `let` interchangeably. I'm wondering what the reasoning behind it is?

So far we've primarily used `let` for variables and `const` for imported modules and function.

Reply:
i prefert `const` over `let` because we prevent accidental overwriting of variables. It is just safer in general IMHO, the reason i sometime end up using let is because of my habit. thanks for pointing it out